cachetools/decorators.py

```diff
@@ -21,10 +21,9 @@ def wrapper(*args, **kwargs):
                     pass  # key not found
                 v = func(*args, **kwargs)
                 try:
-                    cache[k] = v
+                    return cache.setdefault(k, v)
```

Comment:
Arguably, if no `lock` is provided, we might assume this cache instance is not used in multi-threaded context and keep the old behavior, because no cache access should have happened in the mean time.